cli/commands/benchmark-web-vitals.mjs

```diff
@@ -164,13 +164,28 @@ async function benchmarkURL( browser, params ) {
 			);
 
 		// Load the page.
-		const response = await page.goto(
-			`${ params.url }?rnd=${ requestNum }`,
-			{ waitUntil: 'networkidle0' }
-		);
+		const url = new URL( params.url );
+		url.searchParams.append( 'rnd', String( requestNum ) );
+
+		// Make sure any username and password in the URL is passed along for authentication.
+		if ( url.username && url.password ) {
+			await page.authenticate( {
+				username: url.username,
+				password: url.password,
+			} );
+		}
+
+		const response = await page.goto( url.toString(), {
+			waitUntil: 'networkidle0',
+		} );
 		await page.addScriptTag( { content: scriptTag, type: 'module' } );
 
 		if ( response.status() !== 200 ) {
+			log(
+				formats.error(
+					`Request to ${ url } returned status ${ response.status() }.`
+				)
+			);
```

Comment:
Not sure we want the logging here, at least not unconditionally. There's always a chance that a few requests fail, which is why at the end the completion rate is shown, and logging every failed request would clutter the CLI output. If we want to add logging for this, I think it should only be in some `--verbose` mode or something like that.

Reply:
Yeah, it's just that if there is a typo in the URL or something and the completion rate is 0% then there's no indication of what went wrong at present. So then you have to re-request the URL, e.g. via `curl`, to debug further.

Reply:
Maybe we store that error in a variable and only log it if at the end there's a 0% success rate, indicating that something in the actual command usage was wrong?

Reply:
Yeah, that'd work. We can explore that in another PR.